source/arm9/math.c

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Zlib
+//
+// Copyright (C) 2024 Dominik Kurz
+#include <nds/arm9/math.h>
+
+
+float hw_sqrtf(float x)
```

Comment:
I've experimented a bit with this. Can you add `ARM_CODE` in front of this? It seems to save a few CPU cycles. Also add `ARM_CODE` in the header:
```
ARM_CODE float hw_sqrtf(float x)
```